### DISCUSSION TITLE
[Meta] Who is this app for?

### DISCUSSION THREAD

**Discussion by hanaral** [elementary/code#981] [General]
I get that it's very much an IDE, but from my experience it is way too developer-y for normal people. So why is it in a consumer product as the one and only default text editor? Surely a development tool should not be installed by default.
The problem with a developer focused default app isn't just the defaults or the presentation either, it's the fact its for _development_; that it's basically cutting edge with its features.
If it isn't an IDE, then why is the text renderer specifically made for monospace fonts?

I feel that something more akin to TextEdit on mac would be more suitable, that can create and edit RTF files. [Writer](https://github.com/ryonakano/writer) seems to be on the right tracks, and with some polishing and simplification it could easily be a default text editor for elementary OS. 

Anyway despite having my gripes it's still my favourite IDE and I just want to start some discussion around making code more presentable as a product to the average person.

**Comment by jeremypw** [comment]
It might be better/easier to ship a separate app for WYSIWYG text editing rather try and make Code be two different things.  Its name was changed from Scratch to Code to emphasise that it was being developed as an IDE, not a general purposes text editor (in which case elementary currently lacks a native general purposes text editor).  There is at least one 3rd party text editor designed for elementaryos - https://appcenter.elementary.io/com.github.lainsce.quilter/.

**Reply by hanaral** [reply]
Quilter isn't really a general purpose editor - it's for markdown notes
Aaaand it's also now a gnome app and won't be on Odin.
I didn't mean that code should be two different things, that's why I suggested adopting Writer(with some modification). I don't think Code needs to stop being an IDE, there just needs to be a more suitable default text editor.

**Comment by Suzie97** [comment]
I think that providing the option to at least open and read MS Office and Open office document formats would be a nice feature. Almost all consumers users need to deal with documents at least once, and elementaryOS doesn't provide one by default.

Including the option to create Rich Text documents as well as Plain Text would also be useful for general users.

I like how Apple's TextEdit for macOS works. It can display the text in open format documents as well as MS Office documents.

It also has very basic spreadsheets and multi media features like creating simple tables and annotating images.

A similar app for elementaryOS would complement the out of the box experience of elementaryOS.

**Comment by jeremypw** [comment]
I think it unrealistic to hope that elementary devs can produce a MS Office/Libre Office compatible editor in the near future although it is nice goal. Probably an RTF editor would be the first step.

**Reply by Suzie97** [reply]
> 
> 
> I think it unrealistic to hope that elementary devs can produce a MS Office/Libre Office compatible editor in the near future although it is nice goal. Probably an RTF editor would be the first step.

At least opening up MS Office documents with basic syntax might at least offer users some usefulness.

**Reply by jeremypw** [reply]
If there is a suitable library (with Vala bindings preferably) that might be possible.

**Reply by hanaral** [reply]
I'll go ahead and paste some images of TextEdit to see what a normal text editor could look like

**Comment by hanaral** [comment]
Here is an example of TextEdit on macOS Leopard and Writer:

| |  Screenshot 1 | Screenshot 2| Screenshot 3 |
| ---------- | ---------- | ---------- | ---------- | 
| TextEdit in macOS Leopard | ![TextEdit-Leopard](https://user-images.githubusercontent.com/58219504/114007230-af40d780-9850-11eb-9b6c-ce6db73340da.png) | ![TextEdit-Leopard-prefs1](https://user-images.githubusercontent.com/58219504/114007277-b831a900-9850-11eb-97b7-06418d03733c.png) | ![TextEdit-Leopard-prefs2](https://user-images.githubusercontent.com/58219504/114007348-caabe280-9850-11eb-80d4-5d3429df3453.png) |
| Writer for elementary OS | ![Screenshot from 2021-04-08 11-00-40](https://user-images.githubusercontent.com/58219504/114008088-6dfcf780-9851-11eb-9590-e1433b25477c.png) | ![Screenshot from 2021-04-08 11-00-47](https://user-images.githubusercontent.com/58219504/114008121-748b6f00-9851-11eb-9b86-615f91f85b13.png) | ![Screenshot from 2021-04-08 11-01-30](https://user-images.githubusercontent.com/58219504/114008158-7bb27d00-9851-11eb-8f66-d777cbbbd05a.png) |

**Reply by jeremypw** [reply]
I wasn't aware of Writer for elementaryos - is the latest source at https://github.com/btd1337/writer? Because that hasn't been updated for several years.  But it looks as if it might have the right sort of balance between simplicity and functionality for an elementary app (once completed).

**Reply by Suzie97** [reply]
@hanaral has a repo in his profile which is a fork of writer and looks like he is working on it. Are you btw, @hanaral ?

**Reply by hanaral** [reply]
To answer both of your questions, my fork is of @ryonakano's. He's been working to get it functional from time to time, and my fork was just me trying to implement an updated icon.

**Reply by jeremypw** [reply]
If it were accepted as a potential elementary native app I would be willing to work on it but I am quite busy working on Code atm.

**Comment by hanaral** [comment]
I have some new ideas that may solve this issue:
- How about 'Notes' app that finally puts the notes backend in EDS to good use? AFAIK it already supports rich text, and I could easily see a notetaking app that can open and export RTF files!

@jeremypw What are your thoughts?

**Reply by jeremypw** [reply]
Some of us are working on io.elementary.tasks which interacts with eds I believe but that is focussed on tasks and I think only deals in plain text atm.  I think there was a note taking app in the Hera AppCenter by an elementary dev but it isn't in Odin AppCenter yet so I cannot check, but yes a good note taking app, with rich formatting, might be promoted to a native app as Code moves towards a dedicated coding environment.

**Reply by hanaral** [reply]
If you're talking about Notes-Up, yes it's fantastic. But it's markdown, and not a word processor.
I've looked into it, it seems that Fastmail is able to support apple notes by reading and writing rich email-formatted(?) text to an inbox called 'notes' in their DAV backend. If there was an app that natively supported this using the same method on elementary OS, having apple notes and RTF compatibility could literally be the thing that sells it to so many students who want something more than ChromeOS.
@cassidyjames Would you know whether Google Notes has DAV support in the same way as Apple Notes?

**Reply by TeamLinux01** [reply]
Just chiming in here; I would love to see an editor in Elementary OS by default that has good [AsciiDoc](https://asciidoc.org/) support and can export to PDF.
I started using AsciiDoc almost exclusively since I learned about it. Being able to create a Git project and break up different chapters into separate files and use include to bring them all back into one document that I can export as a nice looking PDF is great.
I tried [AsciiDocFX](https://asciidocfx.com/), but I thought it was a bit slow and I wasn't a fan that it required Java.

I currently just use VS Code with the [AsciiDoctor plug-in](https://github.com/asciidoctor/asciidoctor-vscode) to do previewing then use a [docker container of AsciiDoctor](https://github.com/asciidoctor/docker-asciidoctor) to render the PDF; I felt like the plug-in was missing features for the PDF rendering.

**Comment by Suzie97** [comment]
Btw, eds also has contacts support. So maybe a contacts app can also be provided. Where am I supposed to open issues for new app requests??

**Reply by jeremypw** [reply]
Not sure tbh.  I guess you could try opening a feature request here: https://github.com/elementary/triage/issues  but before doing that it would be good to have a fairly well worked out (minimum) specification and justification for the app as a basis for discussion.

**Reply by Suzie97** [reply]
Thanks for the info

**Reply by hanaral** [reply]
[There is already a contacts app being worked on](https://github.com/elementary/friends), although it hardly does anything apart from loading names as it stands.
